src/hotspot/share/opto/mulnode.cpp

```diff
@@ -1296,59 +1308,158 @@ const Type* LShiftLNode::Value(PhaseGVN* phase) const {
   return TypeLong::make( (jlong)r1->get_con() << (jint)shift );
 }
 
-//=============================================================================
-//------------------------------Identity---------------------------------------
-Node* RShiftINode::Identity(PhaseGVN* phase) {
+Node *RShiftNode::IdealIL(PhaseGVN *phase, bool can_reshape, BasicType bt) {
+  // Inputs may be TOP if they are dead.
+  const TypeInteger* t1 = phase->type(in(1))->isa_integer(bt);
+  if (!t1) return NodeSentinel;        // Left input is an integer
```

Comment:
Drive-by: don't use implicit null-check, make comparison with `nullptr` explicit. And add curly braces.